Solutions/Marain.ContentManagement.Rendering.Liquid/Marain/Cms/Internal/LiquidWithMarkdownRenderer.cs

```diff
@@ -0,0 +1,50 @@
+﻿// <copyright file="LiquidWithMarkdownRenderer.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.Cms.Internal
+{
+    using System.IO;
+    using System.Threading.Tasks;
+    using Corvus.Extensions.Json;
+    using DotLiquid;
+    using Markdig;
+
+    /// <summary>
+    /// Writes a content fragment to the output stream.
+    /// </summary>
+    public class LiquidWithMarkdownRenderer : IContentRenderer
+    {
+        /// <summary>
+        /// Gets the registered content type for the renderer.
+        /// </summary>
+        public const string RegisteredContentType = LiquidWithMarkdownPayload.RegisteredContentType + ContentRendererFactory.RendererSuffix;
+
+        private readonly MarkdownPipeline pipeline;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LiquidWithMarkdownRenderer"/> class.
+        /// </summary>
+        /// <param name="pipeline">The pipeline to use for the liquid with markdown rendering.</param>
+        public LiquidWithMarkdownRenderer(MarkdownPipeline pipeline)
+        {
+            this.pipeline = pipeline;
+        }
+
+        /// <summary>
+        /// Gets the content type for the renderer.
+        /// </summary>
+        public string ContentType => RegisteredContentType;
+
+        /// <inheritdoc/>
+        public async Task RenderAsync(TextWriter output, Content parentContent, IContentPayload currentPayload, PropertyBag context)
+        {
+            if (currentPayload is LiquidWithMarkdownPayload liquid)
+            {
+                var template = Template.Parse(liquid.Template);
+                string markdown = await template.RenderAsync(Hash.FromAnonymousObject(new { content = new ContentDrop(parentContent) })).ConfigureAwait(false);
```

Comment:
Yes - this is broadly what it does (I think this is because it has come from whatever platform the Liquid stuff came from originally).

I believe that the reason it is all a bit "odd" is that it has "protect yourself from leaking information" behaviours baked in. Hence this `ContentDrop` type that explicitly exposes a "safe" version of the Content.